### PR TITLE
Add defensive code to the preparation queue

### PR DESCRIPTION
**Pull request by benlodotcom**
- Protection against a nil callback block
- Add a not designated initializer assert

### CODE REVIEW

File: ComponentKit/DataSources/Common/CKComponentPreparationQueue.mm [hunk 1]
```diff
@@ -164,6 +164,11 @@ - (instancetype)initWithQueueWidth:(NSInteger)queueWidth
   return self;
 }
 
+- (instancetype)init 
+{
+  CK_NOT_DESIGNATED_INITIALIZER();
```
Comment (kastiglione) [inline ComponentKit/DataSources/Common/CKComponentPreparationQueue.mm]: Add `CK_NOT_DESIGNATED_INITIALIZER_ATTRIBUTE` in header too?

Comment (benlodotcom) [inline ComponentKit/DataSources/Common/CKComponentPreparationQueue.mm]: Oops had forgotten to add one commit.

**Comment by cfcommando** [comment]
Why not NS_UNAVAILABLE instead?

**Comment by grp** [comment]
One reason: a more semantic error/warning from the compiler.

**Pull request merged**